internal/service/common/utils/repository.go

```diff
@@ -182,16 +183,18 @@ func Create[T db.Model](ctx context.Context, db *pgxpool.Pool, record T) (*T, er
 }
 
 // Update attempts to update a record with a matching primary key.  The stored record is returned on success; otherwise
-// an error is returned.
-func Update[T db.Model](ctx context.Context, db *pgxpool.Pool, record T, uuid uuid.UUID) (*T, error) {
-	tags := GetAllDBTagsFromStruct(record)
+// an error is returned.  If the `fields` argument is set then only those columns are updated but the returned object
+// will contain all columns.
+func Update[T db.Model](ctx context.Context, db *pgxpool.Pool, record T, uuid uuid.UUID, fields ...string) (*T, error) {
+	all := GetAllDBTagsFromStruct(record)
+	tags := GetDBTagsFromStructFields(record, fields...)
```

Comment:
this is how I imaged some specific columns to be selected in Find, FindAll and Search, instead of passing []any for columns

Reply:
ok, I'll rework those other ones in a future PR.